pygmt/tests/test_accessor.py

```diff
@@ -101,3 +103,32 @@ def test_accessor_sliced_datacube():
         assert grid.gmt.gtype == 1  # geographic coordinate type
     finally:
         os.remove(fname)
+
+
+def test_accessor_grid_source_file_not_exist():
+    """
+    Check that the accessor fallbacks to the default registration and gtype
+    when the grid source file (i.e., grid.encoding["source"]) doesn't exists.
+    """
+    # load the 05m earth relief grid, which is stored as tiles
+    grid = load_earth_relief(
+        resolution="05m", region=[0, 5, -5, 5], registration="pixel"
+    )
+    # registration and gtype are correct
+    assert grid.gmt.registration == 1
+    assert grid.gmt.gtype == 1
+    # the source grid file is defined but doesn't exists
+    assert len(grid.encoding["source"]) > 0
```

Comment:
Maybe better to check the file extension is a NetCDF, rather than the length of the filename
```suggestion
    assert grid.encoding["source"].endswith(".nc")
```